wpe/src/main/cpp/browser/entrypoints.cpp

```diff
@@ -193,22 +192,65 @@ void updateAllPageSettings(JNIEnv* env, jclass, jint pageId, jobject jPageSettin
 
     Browser::getInstance().updateAllPageSettings(pageId, settings);
 }
-} // namespace
 
-JNIEXPORT void JNICALL wpe_android_launchProcess(uint64_t pid, wpe::android::ProcessType processType, int* fd)
+// Process management API
+
+struct AndroidProcessProvider {
+    struct wpe_process_provider *wpe_provider;
+};
+
+void* createProcessProvider(struct wpe_process_provider *process_provider)
+{
+    ALOGV("BrowserGlue createProcessProvider()");
+    auto *provider = new AndroidProcessProvider { process_provider };
+    return provider;
+}
+
+void destroyProcessProvider(void* data)
 {
-    if (fd == nullptr) {
-        ALOGE("Cannot launch process (invalid file descriptor)");
+    ALOGV("BrowserGlue destroyProcessProvider()");
+    auto *provider = reinterpret_cast<AndroidProcessProvider*>(data);
+    if (!provider)
         return;
-    }
 
+    delete provider;
```

Comment:
The `delete` operator in C++ is a no-op when applied to a null pointer, so the check here is not really needed. This could be:

```cpp
    delete reinterpret_cast<AndroidProcessProvider*>(data);
```

Reply:
fixed